hydrus/data/crud.py

```diff
@@ -211,20 +210,20 @@ def insert(object_: Dict[str, Any], session: scoped_session,
                     raise NotInstanceProperty(type_=prop_name)
 
             # For insertion in IAC
+            elif session.query(exists().where(RDFClass.name == str(object_[prop_name]))).scalar() \
+                    and property_.type_ == "PROPERTY" or property_.type_ == "ABSTRACT":
+                property_.type_ = "ABSTRACT"
+                session.add(property_)
+                class_ = session.query(RDFClass).filter(
+                    RDFClass.name == object_[prop_name]).one()
+                triple = GraphIAC(
+                    subject=instance.id,
+                    predicate=property_.id,
+                    object_=class_.id)
+                session.add(triple)
             elif session.query(exists().where(RDFClass.name == str(object_[prop_name]))).scalar():

```

Comment:
Why check twice? DB calls are expensive, try and minimize them.

Reply:
Ok, that can only be done in a separate PR, could you make an issue and point out what exactly has to be done?

Reply:
When we had nested `ifs`, the DB condition was only checked once and then the inner conditions are checked. Now because you remove the nesting, we need to check the DB condition for both the inner conditions, this requires 2 DB calls. You can revert the changes or use an intermediate variable that stores the value for the condition. No need for a separate PR though.

Reply:
Ok, that's my fault, I'll get to it as soon as possible 👍